app/src/main/java/com/github/swent/echo/viewmodels/HomeScreenViewModel.kt

```diff
@@ -0,0 +1,167 @@
+                sectionChecked = mutableStateOf(true),
+                classChecked = mutableStateOf(true),
+                pendingChecked = mutableStateOf(true),
+                confirmedChecked = mutableStateOf(true),
+                fullChecked = mutableStateOf(true),
+                from = mutableStateOf(ZonedDateTime.now()),
+                to = mutableStateOf(ZonedDateTime.now()),
+                sortBy = mutableStateOf(SortBy.NONE)
+            )
+        )
+    private val _profileName =
+        MutableStateFlow<String>("John Doe") // placeholder until we have user profiles
+    val profileName = _profileName.asStateFlow()
+    private val _profileClass =
+        MutableStateFlow<String>("IN - BA6") // placeholder until we have user profiles
+    val profileClass = _profileClass.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            val userid = authenticationService.getCurrentUserID()
+            filterTagList =
+                if (userid == null) {
+                    // the user is not logged in, display the default events on map
+                    // TODO add a default tag list
+                    repository.getAllTags().toSet()
+                } else {
+                    // the user is logged in, display the events based on the users tags on map
+                    // repository.getUserProfile(userid).tags
+                    setOf(SAMPLE_TAGS.first())
+                    // SAMPLE_TAGS
+                }
+            allEventsList = SAMPLE_EVENTS // repository.getAllEvents()
+            filterEvents()
+        }
+    }
+
+    suspend fun signOut() {
+        authenticationService.signOut()
+    }
+
+    fun refreshFiltersContainer() {
+        filterTagList =
+            setOf(
+                SAMPLE_TAGS.find { it.name == filtersContainer.value.tagName.value }
+                    ?: Tag("0", "0")
+            ) // replace with the repository call
+        filterEvents()
+    }
+
+    fun resetFiltersContainer() {
+        filtersContainer.value =
+            FiltersContainer(
+                tagName = mutableStateOf(""),
+                epflChecked = mutableStateOf(true),
+                sectionChecked = mutableStateOf(true),
+                classChecked = mutableStateOf(true),
+                pendingChecked = mutableStateOf(true),
+                confirmedChecked = mutableStateOf(true),
+                fullChecked = mutableStateOf(true),
+                from = mutableStateOf(ZonedDateTime.now()),
+                to = mutableStateOf(ZonedDateTime.now()),
+                sortBy = mutableStateOf(SortBy.NONE)
+            )
+        refreshFiltersContainer()
+    }
+
+    // End of methods to set the filters container values
+
+    private fun filterEvents() {
+        _displayEventList.value =
+            allEventsList.filter { event ->
+                event.tags.any { tag -> filterTagList.any { tag2 -> tag.tagId == tag2.tagId } }
```

Comment:
You could use `event.tags.any { tag -> filterTagList.contains(tag) }`. Makes it easier to read... But your solution works as well.

Reply:
I wrote that previously but it didn't seem to work idk why 🧐